### PR TITLE
feat: New Analyzer to avoid cast to `string`

### PR DESCRIPTION
**Pull request by ynsehoornenborg**
Implement, test and doc

### CODE REVIEW

File: Documentation/Diagnostics/PH2142.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# PH2140: Avoid Cast to String
```
Comment (bcollamore) [inline Documentation/Diagnostics/PH2142.md:1]: 2140

File: Documentation/Diagnostics/PH2142.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+| Diagnostic ID | PH2142 |
+| Category  | [Maintainability](../Maintainability.md) |
+| Analyzer | [AvoidCastToStringAnalyzer](https://github.com/philips-software/roslyn-analyzers/blob/main/Philips.CodeAnalysis.MaintainabilityAnalyzers/Maintainability/AvoidCastToStringAnalyzer.cs)
+| CodeFix  | No |
+| Severity | Error |
+| Enabled By Default | Yes |
+
+## Introduction
+
+Avoid casting to int, use `object.ToString()` or a serialization solution instead.
```
Comment (bcollamore) [inline Documentation/Diagnostics/PH2142.md:15]: int?

**Review by bcollamore — CHANGES_REQUESTED** [review]

File: Documentation/Diagnostics/PH2142.md (hunk 2 above)
Comment (ynsehoornenborg) [reply Documentation/Diagnostics/PH2142.md]: oops, typo

File: Documentation/Diagnostics/PH2142.md (hunk 1 above)
Comment (ynsehoornenborg) [reply Documentation/Diagnostics/PH2142.md]: Corrected

**Pull request merged**